lib/auth/index.jsx

```diff
@@ -216,84 +216,73 @@ export class Auth extends Component {
   };
 
   setupAuthWindow = () => {
-    const remote = __non_webpack_require__('electron').remote; // eslint-disable-line no-undef
-    const BrowserWindow = remote.BrowserWindow;
-    const protocol = remote.protocol;
+    // eslint-disable-next-line no-undef
+    const { BrowserWindow, session } = __non_webpack_require__(
+      'electron'
+    ).remote;
+
     this.authWindow = new BrowserWindow({
       width: 640,
       height: 640,
       show: false,
       webPreferences: {
         nodeIntegration: false,
+        session: session.fromPartition(`fresh-session-${Math.random()}`),
       },
     });
 
-    // Register simplenote:// protocol
-    protocol.registerHttpProtocol('simplenote', req => {
-      this.authWindow.loadURL(req.url);
+    this.authWindow.on('closed', () => {
+      // make sure to release this from memory
+      this.authWindow = null;
     });
 
-    this.authWindow.webContents.on('will-navigate', (event, url) =>
-      this.onBrowserNavigate(url)
-    );
-
-    this.authWindow.webContents.on(
-      'did-get-redirect-request',
-      (event, oldUrl, newUrl) => this.onBrowserNavigate(newUrl)
-    );
-  };
-
-  onBrowserNavigate = url => {
-    try {
-      this.authenticateWithUrl(new URL(url));
-    } catch (error) {
-      // Do nothing if Url was invalid
-    }
-  };
-
-  authenticateWithUrl = url => {
-    // Bail out if the url is not the simplenote protocol
-    if (url.protocol !== 'simplenote:') {
-      return;
-    }
-
-    const { authorizeUserWithToken, saveWPToken } = this.props;
-    const params = url.searchParams;
-
-    // Display an error message if authorization failed.
-    if (params.get('error')) {
-      switch (params.get('code')) {
-        case '1':
-          return this.authError(
-            'Please activate your WordPress.com account via email and try again.'
-          );
-        default:
-          return this.authError('An error was encountered while signing in.');
+    this.authWindow.webContents.session.protocol.registerHttpProtocol(
+      'simplenote',
+      (req, callback) => {
+        const { searchParams } = new URL(req.url);
+
+        // cancel the request by running callback() with no parameters
+        // we're going to close the window and continue processing the
+        // information we received in args of the simplenote://auth URL
+        callback();
```

Comment:
So this is what fixed the looping issue?

Reply:
I believe so, that and our manual redirect.